### DISCUSSION TITLE
[Bug]: persistent storages does not mount during build.

### DISCUSSION THREAD

**Discussion by Saifallak** [coollabsio/coolify#2869] [Q&A]
### Description

in nodejs or other scripts too, we might need to mount some config files and .env files and others
but while building those are not mounted, they are mounted after build only.
so we cannot use it and build our scripts correctly.

### Minimal Reproduction (if possible, example repository)

just mount any file/vol. and try to use it in build steps.

### Exception or Error

not found vol/file.

### Version

v4.0.0-beta.315

**Comment by DarkGhostHunter** [comment]
Not a bug, the build is done using an ephemeral container and the Dockerfile has only access to its context and children.

I had the same problem until I noticed the commands. The container that builds the image pulls in the code uses `git clone` into `/artifacts`, then and runs `docker build...` on the `docker-compose.yaml` file. If you have images that require the code, you need to add the context to the service and point to the `Dockerfile`.

```yaml
# ./docker-compose.yaml
service:
  build:
    context: .
    dockerfile: /docker/my-service/Dockerfile
  ...
```

From there, your Dockerfile can pull the code cloned into the container.

```dockerfile
COPY --chown=1000:1000 . /var/www/html
```

**Comment by Saifallak** [comment]
> Not a bug, the build is done using an ephemeral container and the Dockerfile has only access to its context and children.
> 
> I had the same problem until I noticed the commands. The container that builds the image pulls in the code uses `git clone` into `/artifacts`, then and runs `docker build...` on the `docker-compose.yaml` file. If you have images that require the code, you need to add the context to the service and point to the `Dockerfile`.
> 
> ```yaml
> # ./docker-compose.yaml
> service:
>   build:
>     context: .
>     dockerfile: /docker/my-service/Dockerfile
>   ...
> ```
> 
> From there, your Dockerfile can pull the code cloned into the container.
> 
> ```dockerfile
> COPY --chown=1000:1000 . /var/www/html
> ```

It was just simple github repo, not docker image, how to do the same for it?

**Comment by andrasbacsai** [comment]
You should make a custom dockerfile.

Moving this issue to discussions.

**Reply by Saifallak** [reply]
Any example of simple one? ..
Or the one panel using and edit it?
